### PR TITLE
MINOR: closable object Memory leak prevention

### PR DESCRIPTION
**Pull request by highluck**
closable object Memory leak prevention

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: streams/src/main/java/org/apache/kafka/streams/processor/internals/StateDirectory.java [hunk 1]
```diff
@@ -197,8 +197,8 @@ synchronized boolean lock(final TaskId taskId) throws IOException {
 
         final FileChannel channel;
 
-        try {
-            channel = getOrCreateFileChannel(taskId, lockFile.toPath());
+        try (final FileChannel fileChannel = getOrCreateFileChannel(taskId, lockFile.toPath())) {
```
Comment (abbccdda) [inline streams/src/main/java/org/apache/kafka/streams/processor/internals/StateDirectory.java]: It looks weird to use the resource outside of try block, does this really work?

Comment (highluck) [reply streams/src/main/java/org/apache/kafka/streams/processor/internals/StateDirectory.java]: thanks for review 
I was mistaken and wrong

**Comment by highluck** [comment]
@abbccdda 

Please review again thank you :)

**Comment by highluck** [comment]
@abbccdda 
thanks for review 
I was mistaken and wrong

close issue

**Pull request closed without merging**